pyigm/guis/igmguesses.py

```diff
@@ -778,19 +810,25 @@ def add_component(self, inp, vlim=None, zcomp=None, update_model=True):
           False when reading the previous file to increase speed.
         """
         # this is mostly from reading previous IGM_model
+
         if isinstance(inp, AbsComponent):
 
             # get rid of lines outside spectral coverage, if any
             # Todo: one may want to reconsider this, specially if more spectral coverage is obtained
             new_abslines = []
             for absline in inp._abslines:
                 wobs = absline.wrest * (1 + absline.z)
-                if (wobs > self.spec.wvmin) and (wobs < self.spec.wvmax):
+                if (wobs > self.parent.wvmin) and (wobs < self.parent.wvmax):
```

Comment:
Should you be calling the parent `wvmin` or 
one in this object?  Seems to me the latter,
so please double check.

Reply:
Checked again. When igmguesses is used with -p guesses.json -o guesses.json option, calling self.spec.wvmin and self.spec.wvmax deletes absorption lines from guesses file that are not in the spectral range of the input spectrum (i.e., within [self.spec.wvmin, self.spec.wvmax]). On the other hand, calling self.parent.wvmin and self.parent.wvmax includes all absorption lines in a customly defined wavelength range [wmin,wmax]. In this way, one guesses file could be used when working with multiple non-overlapping spectra (or spectra with different wavelength ranges) of the same object. Therefore, it should be self.parent.wvmin and self.parent.wvmax.